common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/ClientCertAuthChallengeHandler.java

```diff
@@ -46,166 +40,65 @@
-    private static final String YUBIKEY_PROVIDER = "YKPiv";
     private final Activity mActivity;
-    private final YubiKitManager mYubiKitManager;
-    private UsbYubiKeyDevice mDevice;
+    //Smartcard CBA variables
+    protected final ISmartcardCertBasedAuthManager mSmartcardCertBasedAuthManager;
     private final DialogHolder mDialogHolder;
-    //Lock to help facilitate synchronization
-    private static final Object sDeviceLock = new Object();
     //Booleans to help determine if a CBA flow is being completed so that we can emit telemetry for the results.
     private boolean mIsOnDeviceCertBasedAuthProceeding;
     private boolean mIsSmartcardCertBasedAuthProceeding;
 
     /**
-     * Holds certificate found on YubiKey and its corresponding slot.
-     */
-    public static class YubiKitCertDetails {
-        private final X509Certificate cert;
-        private final Slot slot;
-
-        /**
-         * Creates new instance of YubiKitCertDetails.
-         * @param cert Certificate found on YubiKey.
-         * @param slot PIV slot on YubiKey where certificate is located.
-         */
-        public YubiKitCertDetails(@NonNull final X509Certificate cert,
-                                  @NonNull final Slot slot) {
-            this.cert = cert;
-            this.slot = slot;
-        }
-
-        /**
-         * Gets certificate.
-         * @return certificate.
-         */
-        @NonNull
-        public X509Certificate getCertificate() {
-            return cert;
-        }
-
-
-        /**
-         * Gets PIV Slot where certificate is located.
-         * @return Slot where certificate is located.
-         */
-        @Nonnull
-        public Slot getSlot() {
-            return slot;
-        }
-    }
-
-    /**
-     * Callback which will contain code to be run upon creation of a PivSession instance.
-     */
-    public interface IPivSessionCallback {
-        /**
-         * Code depending on PivSession instance to be run.
-         * @param piv PivSession instance created from UsbSmartCardConnection.
-         */
-        void onGetSession(@NonNull final PivSession piv);
-    }
-
-    /**
-     * Creates new instance of ClientCertAuthChallengeHandler and starts usb discovery for YubiKeys.
+     * Creates new instance of ClientCertAuthChallengeHandler.
+     * A manager for smartcard CBA is retrieved, and discovery for USB devices is started.
      * @param activity current host activity.
+     * @param smartcardCertBasedAuthManager ISmartcardCertBasedAuthManager instance.
      */
-    public ClientCertAuthChallengeHandler(@NonNull final Activity activity) {
+    public ClientCertAuthChallengeHandler(@NonNull final Activity activity,
+                                          @NonNull final ISmartcardCertBasedAuthManager smartcardCertBasedAuthManager) {
         mActivity = activity;
         mDialogHolder = new DialogHolder(mActivity);
```

Comment:
For testing, I wanted to make sure that `ClientCertAuthChallengeHandler` will bring the correct dialogs up, and I need to be able to interact with the dialogs (type in EditText and see error message disappear, for example), so wouldn't I want the original `DialogHolder` for testing too? Let me know if I'm misunderstanding.
It does make sense though to maybe inject a `DialogHolder` into a smartcard CBA implementation of `ClientCertAuthChallengeHandler` so that the challenge handler doesn't need any sort of activity/context variable.

Reply:
I think I was able to create some unit tests with DialogHolder using ShadowAlertDialog (unless these would be considered UI/Automation tests): https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1839/files
It takes about 9-12 ish seconds to run, so if this can run faster as UI/Automation tests, I can create those along with a DialogHolder interface in a separate PR.